data/crac/crac-impl/src/main/java/com/powsybl/openrao/data/cracimpl/FlowCnecImpl.java

```diff
@@ -176,14 +172,18 @@ public FlowCnecValue computeValue(Network network, Unit unit) {
     }
 
     private double getFlow(Branch branch, TwoSides side, Unit unit) {
-        double power = branch.getTerminal(side).getP();
+        double activeFlow = branch.getTerminal(side).getP();
+        double intensity = branch.getTerminal(side).getI();
         if (unit.equals(Unit.AMPERE)) {
-            power = branch.getTerminal(side).getI();
-            return Double.isNaN(power) ? branch.getTerminal(side).getP() * getFlowUnitMultiplierMegawattToAmpere(side) : power;
+            // In case flows are negative, we shall replace this value by its opposite
+            if (activeFlow < 0) {
+                intensity = -intensity;
+            }
+            return Double.isNaN(intensity) ? activeFlow * getFlowUnitMultiplierMegawattToAmpere(side) : intensity;
```

Comment:
Avoid unnecessary operation in case intensity is nan
```suggestion
            return Double.isNaN(intensity) ? activeFlow * getFlowUnitMultiplierMegawattToAmpere(side) : Math.signum(activeFlow) * intensity;
```